usaspending_api/disaster/v2/views/overview.py

```diff
@@ -101,29 +103,98 @@ def award_outlays(self):
-            .values("gross_outlay_amount_by_tas_cpe", "anticipated_prior_year_obligation_recoveries")
-            .aggregate(
-                total=(Sum("gross_outlay_amount_by_tas_cpe") - Sum("anticipated_prior_year_obligation_recoveries"))
-            )["total"]
-        ) or 0.0
+        return {
+            "total_obligations": results["obligation_totals"] or 0.0,
+            "total_outlays": results["outlay_totals"] or 0.0,
+        }
 
     @staticmethod
     def sum_values(list_of_objects: list, key_to_extract: str) -> Decimal:
         return Decimal(sum([elem[key_to_extract] for elem in list_of_objects]))
+
+    @staticmethod
+    def additional_totals() -> Optional[dict]:
+        """
+        In the case where the filter is for only DEFC "V" this is a special case where some
+        additional values are returned in the response to account for DEFC "V" values
+        that were reported as DEFC "O".
+        """
+        obligation_values = [
+            "obligations_incurred_total_cpe",
+            "deobligations_or_recoveries_or_refunds_from_prior_year_cpe",
+        ]
+        outlay_values = ["gross_outlay_amount_by_tas_cpe", "anticipated_prior_year_obligation_recoveries"]
+        total_budget_authority_values = [
+            "total_budgetary_resources_cpe",
+            "budget_authority_unobligated_balance_brought_forward_cpe",
+            "deobligations_or_recoveries_or_refunds_from_prior_year_cpe",
+            "prior_year_paid_obligation_recoveries",
+        ]
+
+        filters = {
+            "tas_rendering_label__in": ["016-X-0168-000", "016-X-1801-000", "016-X-8042-000"],
+            "disaster_emergency_fund_code": "O",
+            "treasury_account_identifier__funding_toptier_agency__abbreviation": "DOL",
+        }
+
+        aggregates = {
+            "obligation_totals": (
+                Sum("obligations_incurred_total_cpe")
+                - Sum("deobligations_or_recoveries_or_refunds_from_prior_year_cpe")
+            ),
+            "outlay_totals": (
+                Sum("gross_outlay_amount_by_tas_cpe") - Sum("anticipated_prior_year_obligation_recoveries")
+            ),
+            "total_budget_authority": (
+                Sum("total_budgetary_resources_cpe")
+                - (
+                    Sum("budget_authority_unobligated_balance_brought_forward_cpe")
+                    + Sum("deobligations_or_recoveries_or_refunds_from_prior_year_cpe")
+                    + Sum("prior_year_paid_obligation_recoveries")
+                )
+            ),
+        }
+        defc_o_values = (
+            GTASSF133Balances.objects.filter(**filters, fiscal_year=2021, fiscal_period=6)
+            .values(*obligation_values, *outlay_values, *total_budget_authority_values)
+            .aggregate(**aggregates)
+        )
+
+        total_values = (
+            latest_gtas_of_each_year_queryset().filter(**filters, fiscal_year__gte=2021).aggregate(**aggregates)
```

Comment:
Filtering down to the latest period of each fiscal year then filtering for fiscal years gte 2021 will add some unused predicates to the WHERE clause. Although, I doubt this will have much impact on performance

Reply:
Yeh I noticed that in some of the queries I looked at. Thought it was worth it for the sake of more code re-use since performance isn't affected. But can certainly correct it if need be.